Objects/tupleobject.c

```diff
@@ -393,11 +393,17 @@ _PyTuple_FromArray(PyObject *const *src, Py_ssize_t n)
 }
 
 PyObject *
-_PyTuple_FromArraySteal(PyObject *const *src, Py_ssize_t n)
+_PyTuple_FromArraySteal(PyObject *const *ar, Py_ssize_t ar_size, Py_ssize_t start)
```

Comment:
Btw, we might want to tell the compiler to inline this function (if possible), since it's only used in one place anyway.